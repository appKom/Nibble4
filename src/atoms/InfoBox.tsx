import React, { FC, ReactNode } from "react";
import styled from "styled-components";
import { OffWhite, OnlineBlue, WarningOrange } from "utility/style";

export enum InfoBoxTypes {
  DEFAULT = "DEFAULT",
  WARNING = "WARNING",
}

type Props = {
  children: ReactNode;
  type?: InfoBoxTypes;
};

const InfoBox: FC<Props> = ({
  children,
  type = InfoBoxTypes.DEFAULT,
}: Props) => (
  <Container typeColor={type}>
    <p>{children}</p>
  </Container>
);

const Container = styled.div<{ typeColor: InfoBoxTypes }>`
  background-color: ${(props) =>
    props.typeColor === InfoBoxTypes.DEFAULT ? OffWhite : WarningOrange};
  border-radius: 5px;
  padding: 0.5rem 1rem;
  margin: 0.5rem 0;

  span {
    color: ${OnlineBlue};
    text-decoration: underline;
  }
`;

export default InfoBox;
