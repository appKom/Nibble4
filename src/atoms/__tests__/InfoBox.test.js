import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import InfoBox from "../InfoBox";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<InfoBox>Test</InfoBox>);
  expect(asFragment()).toMatchSnapshot();
});
