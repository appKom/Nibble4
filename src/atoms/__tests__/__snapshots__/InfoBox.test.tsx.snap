// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ lbVopv"
  >
    <p>
      Test
    </p>
  </div>
</DocumentFragment>
`;
